packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -192,25 +339,105 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
             "Commitment has expired"
         );
 
+        require(
+            commitment.maxPrincipalPerCollateralAmount == 0 || 
+                _collateralAmount >= getRequiredCollateral(_principalAmount, commitment.maxPrincipalPerCollateralAmount),
```

Comment:
the `0` check is not necessary as the `getRequiredCollateral` would return `0` if `maxPrincipalPerCollateralAmount == 0`